src/view/popup/usePopup.ts

```diff
@@ -36,43 +32,13 @@ export const usePopup = () => {
         setCalculationHistory(calculationsData);
     };
 
-    const setCountryPercentage = async (
-        country: CountryName,
-        percentage: number
-    ) => {
-        await selectedCountriesRepository.setSelectedCountryPercentage(
-            country,
-            percentage
-        );
-        const newMap =
-            await selectedCountriesRepository.getSelectedCountriesAndPercentages();
-        setSelectedCountries(newMap);
-    };
-
-    const sumPercentages = () => {
-        const percentage = Array.from(selectedCountries.values()).reduce(
-            (accumulator, country) => {
-                return accumulator + country;
-            },
-            0
-        );
-
-        if (percentage > 1) {
-            throw new Error(
-                `Error: The sum of the percentages is greater than 100%. Current sum: ${
-                    percentage * 100
-                }%`
-            );
-        }
-
-        return percentage;
-    };
-
     const refreshAndGetSize = async () => {
         try {
-            sumPercentages();
+            selectedCountriesContext.validatePercentages();
```

Comment:
you can pull this value and the one below out of the context to avoid using the full object i.e. `const { selectedCountriesContext: {validatePercentages, selectedCountries} } = useRootContext();`